lib/iris/fileformats/pp.py

```diff
@@ -1642,12 +1641,8 @@ def __eq__(self, other):
             for attr in self.__slots__:
                 attrs = [hasattr(self, attr), hasattr(other, attr)]
                 if all(attrs):
-                    self_attr = getattr(self, attr)
-                    other_attr = getattr(other, attr)
-                    if isinstance(self_attr, biggus.NumpyArrayAdapter):
-                        self_attr = self_attr.concrete
-                    if isinstance(other_attr, biggus.NumpyArrayAdapter):
-                        other_attr = other_attr.concrete
+                    self_attr = as_concrete_data(getattr(self, attr))
+                    other_attr = as_concrete_data(getattr(other, attr))
```

Comment:
@dkillick Yup, I can confirm that `dask` maintains the laziness when comparing lazy objects; it just generates another lazy dask graph, which you need to realize to get the answer. So the following gives a lazy `result`:
```python
result = getattr(self, attr) == getattr(other, attr)
```
And the answer is realized with:
```python
>>> result.compute()
array([ True,  True,  True,  True,  True,  True,  True,  True,  True,  True], dtype=bool)
```
Or indeed, as @pp-mo suggests, the lazy dask result is made concreate by `np.all`:
```python
>>> np.all(result)
True
```